src/test/java/com/google/devtools/build/lib/analysis/VisibilityTest.java

```diff
@@ -202,6 +203,407 @@ public void testConfigSettingVisibilityAlwaysCheckedAtUse() throws Exception {
     assertThat(hasErrors(getConfiguredTarget("//:my_target"))).isFalse();
   }
 
+  @Test
+  public void testAspectImplicitDependencyCheckedAtDefinition_visible() throws Exception {
```

Comment:
Readability: All the tests should be structured as arrange, act, assert, with an empty line between the blocks.

Remove empty lines between `scratch` calls everywhere. And remove empty lines before `useConfiguration` - keep it together with `scratch`.

Reply:
I tried to adapt all tests to that convention, but wasn't sure how to handle `assertThrows`, which is both "act" and "assert".

Reply:
Yeah, we don't have strict guidelines for that. One empty line works. Thanks.